### PR TITLE
Fix update segment when the name is the same [MAILPOET-3343]

### PR DESCRIPTION
**Pull request by lysyjan**
[MAILPOET-3343]

[MAILPOET-3343]: https://mailpoet.atlassian.net/browse/MAILPOET-3343

### CODE REVIEW

File: lib/Segments/SegmentsRepository.php [hunk 1]
```diff
@@ -33,6 +33,23 @@ public function getCountsPerType(): array {
     return $countMap;
   }
 
+  public function isNameUnique(string $name, ?int $id): bool {
```
Comment (costasovo) [inline lib/Segments/SegmentsRepository.php:36]: Does this method have to be public?

**Review by costasovo — COMMENTED** [review]
Looks good! Just a one question.

File: lib/Segments/SegmentsRepository.php (hunk 1 above)
Comment (lysyjan) [reply lib/Segments/SegmentsRepository.php:36]: How can I call this method from the Controller when it is private/protected?

Comment (costasovo) [reply lib/Segments/SegmentsRepository.php:36]: Oh Sorry! My bad! I read the diff as it was one file and thought that it is also in the controller.

**Review by costasovo — APPROVED** [review]

File: lib/Segments/SegmentsRepository.php (hunk 1 above)
Comment (lysyjan) [reply lib/Segments/SegmentsRepository.php:36]: Don't worry. Thank you, that you have looked at this PR during support rotation

**Review by veljkho — APPROVED** [review]

**Pull request merged**